pkg/kafka/auth/tls.go

```diff
@@ -15,57 +15,20 @@
 package auth
 
 import (
-	gotls "crypto/tls"
-	"crypto/x509"
-	"io/ioutil"
-	"path/filepath"
-
 	"github.com/Shopify/sarama"
 	"github.com/pkg/errors"
-)
 
-// TLSConfig describes the configuration properties for TLS Connections to the Kafka Brokers
-type TLSConfig struct {
-	CertPath string
-	KeyPath  string
-	CaPath   string
-}
+	"github.com/jaegertracing/jaeger/pkg/config/tlscfg"
+)
 
-func setTLSConfiguration(config *TLSConfig, saramaConfig *sarama.Config) error {
-	tlsConfig, err := config.getTLS()
-	if err != nil {
-		return errors.Wrap(err, "error loading tls config")
+func setTLSConfiguration(config *tlscfg.Options, saramaConfig *sarama.Config) error {
+	if config.Enabled {
```

Comment:
This is a breaking change: previously, there was no `--kafka.consumer.tls` option (or `producer`), and TLS config would be used on whether the `getTLS()` function returned an error or not. Now, even if all the parameters are set, the collector/ingester won't be using TLS unless a new option is added to the command line, `--kafka.consumer.tls.enabled` (or the one marked as deprecated but that never existed: `--kafka.consumer.tls`).

Reply:
What is the "new" set of parameters to make it work?

Reply:
The `tls.enabled` ones: `kafka.consumer.tls.enabled` and `kafka.producer.tls.enabled`.

Reply:
So to be backward compatible, it would have to use the `getTLS()` approach still - possibly with the `getTLS()` method checking the enabled flag (i.e. if enabled = false then don't use)?

Reply:
That would work, unless "not specified" means "false" when checking the `tls.enabled` flag.

Reply:
Yep - unfortunately `tls.enabled` default is false.

Reply:
Created #2092 to track this.